### ISSUE TITLE
Simulator: are there any keys for changing T5 and T6?

### ISSUE THREAD

**Issue by Alex00101** [FrSkyRC/ETHOS-Feedback-Community#4464]
Thanks

**Comment by Wolepo** [comment]
In general, all the function and trim keys on a German keyboard should be made functional. Apart from the sticks and the switches, nothing works properly.

**Comment by bsongis-frsky** [comment]
[5] and [shift-5]
[6] and [shift-6]
like the other trims

**Comment by Alex00101** [comment]
Thanks,  looking forward

**Comment by Wolepo** [comment]
From which version? That doesn't work for me. Ethos 1.5.15

**Comment by dwheel** [comment]
If you are referring to number keys 5 and 6, it does not work for me either.

**Comment by bsongis** [comment]
They should work in 1.6.2 simulator
And in the nightlies 1.6.3 it should also work in the web simulator:
https://ethos.studio1247.com/nightly16/X18R_FCC

**Comment by bsongis-frsky** [comment]
Would you confirm it works now?

**Comment by dwheel** [comment]
Using X20 pro 1.6.2 simulator and monitoring in the hardware screen. I get no response for T5 and T6 using the number keys 5 and 6.

**Comment by bsongis** [comment]
And it works with number keys 1 to 4?

**Comment by dwheel** [comment]
No. It works for keys 9,0,-,= for trims 1,2,3,4 (maybe not in that order).

**Comment by Alex00101** [comment]
The only keys that are working for me trim wise (1.6.2 on Windows 11) are + - ()
1...6 are not working

**Comment by dwheel** [comment]
Downloaded and installed 1.6.3 a few minutes back. Same results. No response on 1-6.

**Comment by dwheel** [comment]
I tried hitting all of the keys to see if any get a response on T5 or T6, and [Backspace] key works for T5. [Tab] key works for T6. Win 10 American keyboard.

**Comment by Alex00101** [comment]
Yes, Tab and Backspace (with Shift to reverse) work for me as well. Windows 11

**Comment by bsongis** [comment]
Does it work on the online simulator? I hope I can stop releasing the standard simulator soon!
https://ethos.studio1247.com

**Comment by Alex00101** [comment]
> Does it work on the online simulator? I hope I can stop releasing the standard simulator soon! https://ethos.studio1247.com

May I please ask not to stop releasing standard simulator?  I'm concerned it'll make lua development impossible? Like now my IDE automatically pushes all the lua and other files to the sim directory, then launches the simulator,  intercepts debug output, how would it work with web version?

**Comment by bsongis** [comment]
The problem with the simulator is that we need to release N simulators for P protocols, for X platforms. I will think of your working scenario.

**Comment by dwheel** [comment]
Where can I find setup info for the web simulator? For instance, how to move control sticks, how to set up sound, etc. 

    On Thursday, March 20, 2025 at 03:32:01 AM CDT, Bertrand Songis ***@***.***> wrote:  
 
  
The problem with the simulator is that we need to release N simulators for P protocols, for X platforms. I will think of your working scenario.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you commented.Message ID: ***@***.***>
bsongis left a comment (FrSkyRC/ETHOS-Feedback-Community#4464)
The problem with the simulator is that we need to release N simulators for P protocols, for X platforms. I will think of your working scenario.

**Comment by bsongis** [comment]
Sticks are still work in progress. But my question goes to those who cannot move the 5th and 6th trim on the simulator, is it better on the web simulator ?

**Comment by strgaltdel** [comment]
just did a quick&dirty test:

![Image](https://github.com/user-attachments/assets/d7205212-53d5-42a2-842c-728cf8e823d9)

Win10 / Sim 1.62 (x20pro) >> no luck pressing [5] and [shift-5]
WebSim: OK

![Image](https://github.com/user-attachments/assets/69c830b6-a1df-4860-992c-a2aae4ad145a)

**Comment by bsongis** [comment]
Thanks

**Comment by dwheel** [comment]
Yes. Number keys 1 thru 6 control trims 1 thru 6 in web simulator. I'm using win 10 and American keyboard.

Is there a discussion going on about the web simulator anywhere that I could monitor or join into?

**Comment by bsongis** [comment]
You may discuss it everywhere :)
It is far from finished. It raises new ideas everyday

**Comment by dwheel** [comment]
Thanks, I will start here then.

Where do the X20-FCC-WebSimulator.zip; X18-FCC-WebSimulator.zip; etc.; files go after unzipping them?
How do I change from X20 to X18 in the simulator?

**Comment by strgaltdel** [comment]
hmm, this issue was created 7 month ago, and about T5/T6

pretty sure nobody would search here for web sim Q&A's.
(but may be this is exactly what we need)

looks like this sim is somewhere in an early stage and needs some months before going public.
if its really intended to get input at this stage, personally, i would prefer a dedicated issue, category "discussion"

but if there should be not so much "noise" yet, this may be the right place

**Comment by bsongis** [comment]
Not too much noise while it doesn't do 100% of the current simulator ;)
Those files go nowhere.
My working version of the online simulator is there:
https://ethos.studio1247.com
It is a tiny server and it will move to FrSky on due time!

**Comment by strgaltdel** [comment]
ok, in germany we've the phrase 
"let's keep our feet still"

so, let's wait and have a look at your ongoing work,
and let's try to keep the number of questions and requests as low as possible ,

and let's communicate here....


but there will be noise (-;
https://www.rcgroups.com/forums/showpost.php?p=53615081&postcount=34992
